pints/_mcmc/_dual_averaging.py

```diff
@@ -0,0 +1,241 @@
+#
+# Dual Averaging step size and mass matrix adaption method for NUTS and HMC
+# samplers
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+
+
+class DualAveragingAdaption:
+    """
+    Implements a Dual Averaging scheme to adapt the step size ``epsilon``, as
+    per [1]_, and estimates the (fully dense) inverse mass matrix using the
+    sample covariance of the accepted parameter, as suggested in [2]_
+
```

Comment:
I've added in pseudocode here -- could you check the h_bar equation;  in particular, the use of alpha / n_alpha in it? This was following algorithm 6 from the Hoffman paper...

Reply:
It looks, on the face of it, slightly different to yours...?

Reply:
`n/n_\alpha` has no meaning in this context, this term represented the acceptance probability reported by the algorithm (HMC or Nuts). I've updated the pseudocode

Reply:
Cool, thanks.